tools/deprecate_version/deprecate_version.py

```diff
@@ -151,38 +160,37 @@ def GetRuntimeAndPr():
 
   # grep source code looking for reloadable features which are true to find the
   # PR they were added.
-  grep_output = subprocess.check_output('grep -r "envoy.reloadable_features\." source/', shell=True)
   features_to_flip = []
-  runtime_feature_regex = re.compile(r'.*(source.*cc).*"(envoy.reloadable_features\.[^"]+)".*')
-  for line in grep_output.splitlines():
-    match = runtime_feature_regex.match(str(line))
-    if match:
-      filename = (match.group(1))
-      runtime_guard = match.group(2)
-      # If this runtime guard isn't true, ignore it for this release.
-      if not runtime_guard in runtime_already_true:
-        continue
 
-      # For true runtime guards, walk the blame of the file they were added to,
-      # to find the pr the feature was added.
-      for commit, lines in repo.blame('HEAD', filename):
-        for line in lines:
-          if runtime_guard in line:
-            pr = (int(re.search('\(#(\d+)\)', commit.message).group(1)))
-            # Add the runtime guard and PR to the list to file issues about.
-            features_to_flip.append((runtime_guard, pr))
-            # Make sure if grep finds multiple spots we only do the work one time.
-            runtime_already_true.remove(runtime_guard)
+  runtime_features = re.compile(r'.*"(envoy.reloadable_features..*)",.*')
 
-    else:
-      print('no match in ' + str(line) + ' please address manually!')
+  removal_date = date.today() - datetime.timedelta(days=183)
 
+  # Walk the blame of runtime_features and look for true flags older than 6 months.
+  for commit, lines in repo.blame('HEAD', 'source/common/runtime/runtime_features.cc'):
```

Comment:
I wonder if the previous function should return a tuple of the flag and metadata so we don't need to walk the file again here and here we only check dates and make safe to remove decisions.

Reply:
That's a really good point - we used to have to grep everything to find the false guards but now they're in one file this can totally be cleaned up a ton.  thanks!